src/commands/reword.rs

```diff
@@ -0,0 +1,354 @@
+    let commits = match commits {
+        ResolveCommitsResult::Ok { commits } => commits,
+        ResolveCommitsResult::CommitNotFound { commit: hash } => {
+            writeln!(effects.get_output_stream(), "Commit not found: {}", hash)?;
+            return Ok(None);
+        }
+    };
+    Ok(Some(commits))
+}
+
+/// The result of building the reword message.
+#[must_use]
+pub enum BuildRewordMessageResult {
+    /// The reworded message was built successfully.
+    Succeeded {
+        /// The reworded messages for each commit.
+        messages: HashMap<NonZeroOid, String>,
+    },
+
+    /// The reworded message matches the original message.
+    IdenticalMessage,
+
+    /// The reworded message was empty.
+    EmptyMessage,
+}
+
+/// Builds the message(s) that will be used for rewording. These are mapped from each commit's
+/// NonZeroOid to the relevant message.
+#[instrument]
+fn build_messages(
+    messages: &[String],
+    commits: &[Commit],
+    comment_char: Option<u8>,
+) -> eyre::Result<BuildRewordMessageResult> {
+    let message = messages.join("\n\n").trim().to_string();
+
+    let (message, load_editor) = if message.is_empty() {
+        let message_for_editor = match commits.to_vec().as_slice() {
+            [commit] => match commit.get_message_raw()?.into_string() {
+                Ok(msg) => msg,
+                _ => eyre::bail!(
+                    "###
+                    Error decoding original message for commit: {:?}.
+                    Stopping; nothing reworded.
+                    ###",
+                    commit.get_oid()
+                ),
+            },
+            _ => {
+                // TODO(bulk edit) build a bulk edit message for multiple commits
+                format!(
+                    "Enter a commit message to apply to {} commits",
+                    commits.len()
+                )
+            }
+        };
+        (message_for_editor, true)
+    } else {
+        (message, false)
+    };
+
+    let message = if load_editor {
+        // Editor::edit() normally requires that the file being editted is saved before the editor
+        // is closed. If it's not, it will return None; and in all other cases, it will return
+        // Some(message). We don't care if the file has been saved, only if it hasn't changed, so
+        // here we call `require_save(false)` and just ignore the None case.
+        let edited_message = Editor::new()
+            .require_save(false)
+            .edit(message.as_str())?
+            .unwrap();
+        if edited_message == message {
+            return Ok(BuildRewordMessageResult::IdenticalMessage);
+        }
+        edited_message
+    } else {
+        message
+    };
+
+    // clean up the commit message: remove comment lines, finish w/ a newline, etc
+    // TODO ask if this should be here or if it should be moved to the git module per project wiki
+    let message = git2::message_prettify(message, comment_char)?;
```

Comment:
2 things:

1. The [wiki](https://github.com/arxanas/git-branchless/wiki/Coding#the-git-module) states that git2 code should not be used directly. Would you prefer that I move this into our own git module, or is this trivial enough that it can probably stay here?
2. By doing this processing here, vs up a few lines after the Editor returns, it means that we'll be prettifing *all* messages, including those submitting on the CLI. For example, `git reword -m '#123 is fixed'` will be removed and the user will get an "empty message" notice that nothing was reworded. I *think* that this is OK for consistency and simplicity, but I do have a feeling that suggests that if someone puts a message like that in via CLI that they really intend for it work. (Even though such a message in the editor would not work.) It would be easy enough (I think) to move this, but it does mean that we won't be able to test this. (We currently aren't testing the Editor code path.)

Reply:
> Would you prefer that I move this into our own git module

Let's move this into our own `git` module. In particular, the wrapper should have `#[instrument]`, or else it won't appear in span-traces.

> It would be easy enough (I think) to move this, but it does mean that we won't be able to test this

I think that's fine.

Reply:
OK. I moved that into into the `repo` mod, which was the closest/most relevant that I could find. Happy to move elsewhere if that's not right. This also gave me the chance to update the `get_comment_char` to return a `char`, which feels more natural than the funny feeling (to me) `Option<u8>` that git2 was using.